backend/user_service/user/app/user_application_service.py

```diff
@@ -34,14 +36,23 @@ def register(self, email, password, user_type, car_type, plate_no):
     def login(self, user_id):
         user = get_user_model().objects.get(id=user_id)
         user_type = user.user_type
+
         if user_type == "rider":
-            if len(Rider.objects.filter(user_id=user_id)) == 0:
-                return Rider.objects.create(user=user, status="IDLE")
+            rider = None
+            if(len(Rider.objects.filter(user_id=user_id)) == 0):
+                rider = Rider.objects.create(user=user, status="IDLE")
+            else:
+                rider = Rider.objects.get(user=user)
+            return RiderSerializer(rider).data
+
         elif user_type == "driver":
-            if len(Driver.objects.filter(user_id=user_id)) == 0:
-                return Driver.objects.create(user=user, status="IDLE")
+            driver = None
+            if(len(Driver.objects.filter(user_id=user_id)) == 0):
+                driver = Driver.objects.create(user=user, status="IDLE")
+            else:
+                driver = Rider.objects.get(user=user)
+            return DriverSerializer(driver).data
```

Comment:
I think we can refactor this application logic. I'll create issue to refactor this logic!